### PR TITLE
Configure Base URL for Github Client

### PR DESCRIPTION
**Pull request by alan20854**
adds the ability to write to a specified url instead of the default https://api.github.com/

Looking at https://github.com/google/go-github/issues/958, looks like the url needs to be suffixed by `/api/v3/`

### CODE REVIEW

File: .github/workflows/nightfalldlp.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     steps:
       - name: Add $GOPATH/bin
         run: |
-          echo ::add-path::$(go env GOPATH)/bin
+          echo $(go env GOPATH)/bin >> $GITHUB_PATH
```
Comment (alan20854) [inline .github/workflows/nightfalldlp.yml:14]: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#adding-a-system-path

https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
Screen Shot 2021-03-10 at 11 13 06 AM

**Comment by alan20854** [comment]
How can I test this for enterprise?

File: cmd/nightfalldlp/main.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ const (
 	nightfallConfigFileName = ".nightfalldlp/config.json"
 	githubActionsEnvVar     = "GITHUB_ACTIONS"
 	githubTokenEnvVar       = "GITHUB_TOKEN"
+	githubBaseUrlEnvVar     = "BASE_URL"
```
Comment (alan20854) [inline cmd/nightfalldlp/main.go]: TODO: update readme docs on https://github.com/nightfallai/nightfall_dlp_action and https://github.com/nightfallai/nightfall_circle_orb on how to use this

Comment (rssathe) [inline cmd/nightfalldlp/main.go]: Shouldn't this be GITHUB_API_BASE_URL

Comment (rssathe) [inline cmd/nightfalldlp/main.go]: likewise for the variable Name

File: cmd/nightfalldlp/main.go [hunk 3]
```diff
@@ -87,15 +88,17 @@ func CreateDiffReviewerClient() (diffreviewer.DiffReviewer, error) {
 		if !ok {
 			return nil, fmt.Errorf("could not find required %s environment variable", githubTokenEnvVar)
 		}
-		return github.NewAuthenticatedGithubService(githubToken), nil
+		baseUrl, _ := os.LookupEnv(githubBaseUrlEnvVar)
```
Comment (rssathe) [inline cmd/nightfalldlp/main.go]: Can't you jsut create this once above the switch case?

File: internal/clients/diffreviewer/github/github_client.go [hunk 4]
```diff
@@ -14,13 +16,24 @@ type Client struct {
...
 	ctx := context.Background()
 	ts := oauth2.StaticTokenSource(
 		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(ctx, ts)
 	githubClient := github.NewClient(tc)
+	// for enterprise
+	if baseUrl != "" {
+		u, _ := url.Parse(baseUrl)
+		if !strings.HasSuffix(u.Path, "/") {
```
Comment (rssathe) [inline internal/clients/diffreviewer/github/github_client.go:29]: is it not worth just overwriting the path entirely with fmt.String("%v%v", u.HostName, githubAPIPath ). Do we want them to customize this. Seems like we should ensure it's always v3?

**Review by rssathe — APPROVED** [review]
Nits

File: internal/clients/diffreviewer/github/github_client.go (hunk 4 above)
Comment (alan20854) [reply internal/clients/diffreviewer/github/github_client.go:29]: if they don't specify a protocol, only `u.Path` has a value, the full path including the suffixes, and u.Host is empty. If we want to go the suggested route ^, we'd have to specifically tell users to prepend the protocol when setting their base url.

**Pull request merged**